File: 16주차_프로그래머스_lv1,2/155652.py
def solution(s, skip, index):
    ans = []
    arr = []
    for i in range(26):
        c = chr(97+i)
        if c not in skip:
            arr.append(c)
            
    for i in s:
        ans.append(arr[(arr.index(i)+index) % len(arr)])
    ans = ''.join(ans)
    return ans
# 둘만의 암호
# 배열.index(문자) => 그 배열 내에서 해당 문자의 인덱스 반환. 중복된 문자 일시, 제일 앞 인덱스 반환.

File: 2025/03/155652.js
function solution(s, skip, index) {
  var answer = "";
  const filtered = [];
  let alpha = 97;
  const arr = new Array(26).fill(0);
  arr.forEach((v) => {
    const str = String.fromCharCode(alpha);
    if (!skip.includes(str)) filtered.push(str);
    alpha += 1;
  });

  const strArr = [...s];
  strArr.forEach((v) => {
    const idx = filtered.findIndex((value) => value === v);
    const targetIdx = (idx + index) % filtered.length;
    answer += filtered[targetIdx];
  });

  return answer;
}
// 프로그래머스 둘만의 암호
